reporting_app/static/library_preps.js

```diff
@@ -0,0 +1,202 @@
+    'ntp_volume': {name: 'NTP Volume (uL)', path: ['udf', 'NTP Volume (uL)']},
+    'adjusted_conc': {name: 'Adjusted Conc. (nM)', path: ['udf', 'Adjusted Conc. (nM)']},
+    'original_conc': {name: 'Original Conc. (nM)', path: ['udf', 'Original Conc. (nM)']},
+    'ntp_transfer_vol': {name: 'NTP Transfer Volume (uL)', path: ['udf', 'NTP Transfer Volume (uL)']},
+    'rsb_transfer_vol': {name: 'RSB Transfer Volume (uL)', path: ['udf', 'RSB Transfer Volume (uL)']},
+    'sample_transfer_vol': {name: 'Sample Transfer Volume (uL)', path: ['udf', 'Sample Transfer Volume (uL)']},
+    'tsp1_transfer_vol': {name: 'TSP1 Transfer Volume (uL)', path: ['udf', 'TSP1 Transfer Volume (uL)']},
+    'qc_flag': {name: 'QC flag', path: ['qc_flag'], categories: ['UNKNOWN', 'PASSED', 'FAILED', 'ERRORED']}
+};
+
+
+function get_lims_and_qc_data(lims_url, qc_url, token, library_id) {
+    // query the Lims library_info endpoint for a single library, set library_data and merge in data from samples endpoint
+    $.ajax(
+        {
+            url: lims_url + '?match={"library_id":"' + library_id + '"}',
+            headers: {'Authorization': token},
+            dataType: 'json',
+            success: function(result) {
+                library_data = result.data[0];
+
+                var sample_queries = [];  // sample IDs to merge on
+                var sample_coords = {};  // map sample IDs to well coordinates so we know where to merge the data
+
+                var coords = Object.keys(library_data['qc']);
+                for (var i=0; i<coords.length; i++) {
+                    var coord = coords[i];
+                    var placement = library_data['qc'][coord];
+                    var sample_id = placement['name'];
+                    sample_coords[sample_id] = coord;
+                    sample_queries.push('{"sample_id":"' + sample_id + '"}');
+                }
+
+                // query the samples endpoint for IDs found above, merge in the data and trigger the chart
+                $.ajax(
+                    {
+                        url: qc_url + '?where={"$or":[' + sample_queries.join(',') + ']}&max_results=1000',
+                        headers: {'Authorization': token},
+                        dataType: 'json',
+                        success: function(result) {
+                            var j;
+                            var nsamples = result.data.length;
+                            for (j=0; j<nsamples; j++) {
+                                var sample = result.data[j];
+                                var coord = sample_coords[sample['sample_id']];
+                                library_data['qc'][coord]['reporting_app'] = sample;
+                            }
+                            trigger(active_colour_metric);
+                        }
+                    }
+                );
+            }
+        }
+    );
+}
+
+function query_nested_object(top_level, query) {
```

Comment:
I think lodash get function can do the same thing
https://lodash.com/docs/4.17.11#get